### ISSUE TITLE
Sender address with mailextensions not possible

### ISSUE THREAD

**Issue by Jan1205** [PHPMailer/PHPMailer#2656]
The following mail address is valid: mailbox+test@example.com
PHPMailer does not accept this address as sender address.

In line 1820 is a comment of the usage of + in sender address.
https://github.com/PHPMailer/PHPMailer/blob/master/src/PHPMailer.php#L1820

In line 1823 the check of the characters does not allow the +.
https://github.com/PHPMailer/PHPMailer/blob/master/src/PHPMailer.php#L1823

Is it possibile to add the + in check?

old (line 1823):
`            if (!ctype_alnum($c) && strpos('@_-.', $c) === false) {
                return false;
            }`

new (line 1823):
`            if (!ctype_alnum($c) && strpos('@_-.+', $c) === false) {
                return false;
            }`

The change fixes the issue.

More details:
https://github.com/swiftmailer/swiftmailer/commit/e6ccf40d856af9598b76eb313b215eed25ae9e86
https://gist.github.com/Zenexer/40d02da5e07f151adeaeeaa11af9ab36

**Comment by Synchro** [comment]
I'm _very_ familiar with this. PHPMailer *will* accept `+` in email addresses, but not if you are sending via `mail()`, because it's not possible to know whether doing so is safe. This is because some shells/locales use `+` as an escape character. There is no way to tell if a character is safe to use in a shell command because PHP's `escapeshellarg` and `escapeshellcmd` functions are not aware of the shell's configuration, and they make certain assumptions that may not be true (which is one reason those functions have a poor security track record). Since the reason for the introduction of this limitation in PHPMailer was a major security hole, it's not being reinstated.

In short, use SMTP to localhost, and you'll be able to use `+` in your email addresses.
